File: 0402 - Remove K Digits/solution.py
class Solution:
    def removeKdigits(self, num: str, k: int) -> str:
        res = []
        for c in num:
            while k and res and res[-1]>c:
                res.pop(); k -= 1
            if res or c!='0':
                res.append(c)
        while res and k:
            res.pop(); k -= 1
        return ''.join(res) or '0'

File: 0402 - Remove K Digits/solution.cpp
class Solution {
public:
    string removeKdigits(string num, int k) {
        string res = "";
        for(auto c : num){
            while(k && res!="" && res.back()>c) res.pop_back(), k--;
            if(res!="" || c!='0') res.push_back(c);
        }
        while(res!="" && k--) res.pop_back();
        return res==""?"0":res;
    }
};